radixdlt-java/src/main/java/com/radixdlt/client/core/network/RadixJsonRpcClient.java

```diff
@@ -77,19 +72,23 @@ public RadixJsonRpcClient(WebSocketClient wsClient) {
 			.publish()
 			.refCount();
 
-		if (!CHECK_API_VERSION) {
-			this.serverApiVersion = Single.just(API_VERSION);
-		} else {
-			this.serverApiVersion = jsonRpcCall("Api.getVersion")
-				.map(result -> result.getAsJsonObject().get("version").getAsInt())
-				.cache();
-		}
+		this.serverApiVersion = SingleSubject.create();
+		this.universeConfig = SingleSubject.create();
+	}
 
+	void fetchClientInformation() {
 		Serialization serialization = Serialize.getInstance();
-		this.universeConfig = jsonRpcCall("Universe.getUniverse")
+		jsonRpcCall("Api.getVersion")
+				.map(result -> result.getAsJsonObject().get("version").getAsInt())
+				.onErrorReturn(e -> {
+					LOGGER.error(String.format("Error while requesting Api.getVersion: %s", e));
+					return API_VERSION; // assume api version matches for now until fixed in core
+				})
+				.subscribe(serverApiVersion::onSuccess);
```

Comment:
Still not sure why original method was removed. Doing it this way makes you lose error information in your serverApiVersion subject. Subjects should be used with care in RX and only when necessary as they add a lot of state management: https://medium.com/upday-devs/subjects-neither-good-nor-evil-8384ad23a2ef